{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2/8/16 BUG meeting \n",
    "\n",
    "---\n",
    "\n",
    "### Terry Gaasterland: \n",
    "**Intro to flexible command-line biocomputing**\n",
    "\n",
    "Basic command line intro (cd, ls, mkdir, clear, touch, ...)\n",
    "\n",
    "#### What about more power- using more cores\n",
    "Triton:\n",
    "300-some nodes with 16-24 cores each\n",
    "\n",
    "e.g.\n",
    "\n",
    ">ssh tscc-login1.sdsc.edu -l tg\n",
    "\n",
    "/oasis is a directory with 100 TB \"scratch\" space- \n",
    "\n",
    "file can be \"renewed\":\n",
    "> touch \n",
    "\n",
    "files can be transfered by scp to push or pull data onto triton\n",
    "\n",
    "Moving data around can be the most cumbersome of\n",
    "\n",
    "example using triton to blast a dataset of contigs against each other\n",
    "\n",
    "grep excluding line headers and end-of-line characters\n",
    "\n",
    "### biotools\n",
    "If there is software of interest, it could be irequested from \n",
    "\n",
    "module load biotools\n",
    "pushd /opt/biotools\n",
    "\n",
    "module whatis biotools # shows everything that has been installed\n",
    "\n",
    "includes lots of \n",
    "- fastqc\n",
    "- qiime1.8.0\n",
    "- trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "biotools             :                        biotools\n",
    "biotools             : bedtools 2.20.1        bamtools 2.3.0          biopython 1.62\n",
    "biotools             : bismark 0.10.1         blast 2.2.29            blat 35\n",
    "biotools             : bowtie 1.0.0           bowtie2 2.2.3           bwa 0.7.5a\n",
    "biotools             : bx-python 0.7.2        cufflinks 2.2.0         dendropy 3.12.0\n",
    "biotools             : edena 3.131028         fastqc 0.10.1           fastx 0.0.14\n",
    "biotools             : GenomeAnalysisTK 2.8.1 gmap_gsnap 2014_02_20   htseq 0.5.4p5\n",
    "biotools             : idba-ud 1.1.1          matt 1.00               miRDeep2 0.0.5\n",
    "biotools             : miso 0.5.2             picard 1.108            plink 1.07\n",
    "biotools             : pysam 0.7.8pre         qiime 1.8.0             randfold 2.0\n",
    "biotools             : rsqec 2.4              samtools 0.1.19         soapdenovo r240\n",
    "biotools             : SOAPsnp v1.03          spades 3.1.0            squid 1.9g\n",
    "biotools             : stacks 1.20            tophat 2.0.10           trimmomatic 0.32\n",
    "biotools             : trinity 20131110       velvet 1.2.10           ViennaRNA 2.1.7"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <font color=\"red\"> Followup after meeting </font>\n",
    "\n",
    "From Terry's half\n",
    "1. Will send out a link to get a triton login\n",
    "   - certain number of cycles for free\n",
    "   - is under-utilized by the campus\n",
    "2. Scripps genome center owns several nodes, and therefore thousands of cycles which could be available to you\n",
    "\n",
    "From Tessa's half\n",
    "1. Where to find program MMT\n",
    "1. Where to find supervisor script\n",
    "1. Where to find the tutorial\n",
    "1. post presentation slides\n",
    "\n",
    "1. send another note about the release!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Tessa pierce\n",
    "**Make My Transcriptome: an automated pipeline for de novo transcriptomics**\n",
    "\n",
    "past difficulties\n",
    "- installing software\n",
    "- running software\n",
    "\n",
    "Goal: zero-entry analysis through 1) training and programming 2) simpler software\n",
    "\n",
    "Research: de novo assembly of \n",
    "\n",
    "\n",
    "**Automated via python \"SUpERVISOR\"** Job manager for parallel\n",
    "- lightweight python automation\n",
    "   1. command\n",
    "   1. dependencies\n",
    "   1. targets\n",
    "   1. (optional stuff) name, location, # processors, email for updates\n",
    "\n",
    "Specific application in MMT\n",
    "- quick start shows commands, making .csv file\n",
    "- tutorial online\n",
    "- helper modules \n",
    "\n",
    "MMT is capable of \n",
    "- assembly\n",
    "- annotation\n",
    "- differential exression\n",
    "\n",
    "RNAseq data\n",
    "Evaluating data quality\n",
    "    - length based measures (N50)\n",
    "    - Annotation based (e.g. genome completeness)\n",
    "Assembly quality\n",
    "    - many different means\n",
    "Choosing methods- it matters:\n",
    "    - saw that Trinity works best\n",
    "    - length is not a good metric of number of ORFs, especially with RNA\n",
    "        - annotation metrics are better\n",
    "How much to sequence\n",
    "    - the more complex a sample, the more reads?  Tessa's findings...\n",
    "    - digitally normalize large datasets\n",
    "pE v SE\n",
    "    - No SE!!  bad!!\n",
    "Read Length\n",
    "    - unclear what is best for RNA dataset\n",
    "    - insert length has greater effect\n",
    "    \n",
    "Advantage to using MMT:\n",
    "    - Ease of use\n",
    "    - Ease of subsitution \n",
    "    - Clear starting parameters\n",
    "    - Good Logging"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
